java/core/src/java/org/apache/orc/impl/SchemaEvolution.java

```diff
@@ -513,7 +513,7 @@ void buildConversion(TypeDescription fileType,
             isOnlyImplicitConversion = false;
           }
 
-          if (positionalLevels == 0) {
+          if (positionalLevels <= 0) {
```

Comment:
This can be negative because of ORC-667.

Reply:
because of ORC-644

Reply:
Oh. Thanks! It's fixed.